node/resolvers/search/product.ts

```diff
@@ -328,5 +328,7 @@ export const resolvers = {
   },
   OnlyProduct: {
     categoryTree: productCategoriesToCategoryTree,
+    productClusters: ({ productClusters }: any) => productClusters ?? [],
+    clusterHighlights: ({ clusterHighlights }: any) => clusterHighlights ?? [],
```

Comment:
```suggestion
    productClusters: ({ productClusters }: any) => ([] as any[]).concat(productClusters ?? []),
    clusterHighlights: ({ clusterHighlights }: any) => ([] as any[]).concat(clusterHighlights ?? []),
```

this change fixes the issues I was having (`clusterHighlights` did exist but it was an empty object, not iterable, but this fallback didnt kick in)